mk/lib.mk

```diff
@@ -97,6 +98,16 @@ $(foreach test-group, $(install-tests-groups), \
     $(eval $(call run-test,$(test),$(install_test_init))) \
     $(eval $(test-group).test-group: $(test).test)))
 
+# Compilation database.
+$(foreach lib, $(libraries), $(eval $(call write-compile-commands,$(lib))))
+$(foreach prog, $(programs), $(eval $(call write-compile-commands,$(prog))))
+
+compile_commands.json: $(compile-commands-json-files)
+	@jq --slurp '.' $^ >$@
+
+clean-compile-commands-json:
+	$(suppress) rm -fv -- compile_commands.json $(compile-commands-json-files)
+
```

Comment:
What's the use of this? That should already be covered by `make clean`, right?

Reply:
Removed :relaxed: